Client/Extensions/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,146 @@
+                        guard let self = self else {
+                            return .flyDown
+                        }
+                        guard let rewards = self.tab?.rewards, let publisher = self.tab?.publisher, let amount = Double(details.total.amount.value) else {
+                            return .flyDown
+                        }
+                        
+                        rewards.ledger.tipPublisherDirectly(publisher, amount: amount, currency: "BAT") { _ in
+                          // TODO: Handle started tip process
+                            self.response = """
+                                {
+                                  "requestId": "a62c29b3-f840-47cd-b895-4573d3190227",
+                                  "methodName": "bat",
+                                  "details": {
+                                    "transaction_id": "bcbbd947-346d-439f-96b4-101bbd966675",
+                                    "message": "Payment for Ethiopian Coffee!"
+                                  }
+                                }
+                            """
+                            
+                            ensureMainThread {                               
+                                let trimmed = self.response.removingNewlines()
+                                self.tab?.webView?.evaluateJavaScript("PaymentRequestCallback\(self.token).paymentreq_postCreate('\(trimmed)', '', '')", completionHandler: { _, error in
+                                        if error != nil {
+                                            log.error(error)
+                                        }
+                                    })
+                            }
+                        }
+                        
+                        return .flyDown
+                    }
+                    
+                    popup.addButton(title: Strings.paymentRequestCancel) { [weak self] in
+                        guard let self = self else {
+                            return .flyDown
+                        }
+                        
+                        ensureMainThread {
+                            self.sendPaymentRequestError(errorName: Strings.abortErrorName, errorMessage: Strings.userCancelledMessage)
+                        }
+                        
+                        return .flyDown
+                    }
+                    
+                    log.info("Success!")
+                }
+            } catch {
+                log.info(error)
+            }
+            popup.showWithType(showType: .flyUp)
+        }
+    }
+}
+
+extension String {
+    func removingNewlines() -> String {
+        return components(separatedBy: .newlines).joined()
+    }
+}
+
+extension Strings {
+    public static let paymentRequestTitle = NSLocalizedString("paymentRequestTitle", tableName: "BraveShared", bundle: Bundle.braveShared, value: "Review your payment", comment: "Title for Brave Payments")
```

Comment:
These all probably need better comments. A translator will have no idea what Brave Payments, or "Payment Request screen is".. or especially what "DOMException for AbortError" is.

Maybe not clear, but these comments are for translators, not developers.